sched/jobDefSerializer.go

```diff
@@ -0,0 +1,112 @@
+package sched
+
+import (
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/scootdev/scoot/sched/gen-go/sagalog"
+	"github.com/scootdev/scoot/runner"
+	"time"
+	"fmt"
+)
+
+const (
+	JsonSerialize string = "json"
+	BinarySerialize string = "binary"
+)
+
+func Deserialize(serializedVal []byte, serializeType string) (*JobDefinition, error) {
+
+	// create the thrift deserializer
+	//deserializer := thrift.NewTDeserializer();
+	deserializer, err := getDeserializer(serializeType)
+	if (err != nil) {
+		return nil, err
+	}
+
+	// create the thrift jobDef struct for the saga log
+	sagaLogJobDef := sagalog.NewJobDefinition()
+	// parse the byte string into the thrift jobDef struct
+	if err := deserializer.Read(sagaLogJobDef, serializedVal); err != nil {
+		return nil, err
+	}
+
+	// make a sched JobDefintion from the thrift JobDefinition for the saga log
```

Comment:
This seems like a great thing to hoist into a function...

In fact, it might be in scootapi/server/serve.go (in thriftJobToScoot)?